src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -540,10 +544,33 @@ public string Call(JObject opts, string? blockId)
                 : throw new Exception("Error in contract call");
         }
 
+        private bool IsGreater(ulong? lastBlockGasUsed, ulong percent, ulong gasLimit)
+        {
+            return lastBlockGasUsed * 100 > percent * gasLimit;
+        }
+
         [JsonRpcMethod("eth_gasPrice")]
         public string GetNetworkGasPrice()
         {
-            return Web3DataFormatUtils.Web3Number(_stateManager.CurrentSnapshot.NetworkGasPrice.ToUInt256());
+            var gasLimit = GasMetering.DefaultBlockGasLimit;
+
+            var latestBlock = _blockManager.LatestBlock();
+            var previousBlockHash = latestBlock.Hash.ToHex();
+            var lastBlockGasUsed = GetGasUsed(previousBlockHash);
+
+            Logger.LogInformation($"lastBlockGasUsed:: {lastBlockGasUsed}");
+            Logger.LogInformation($"previousBlockHash:: {previousBlockHash}");
+
+            if (IsGreater(lastBlockGasUsed, 10, gasLimit))
```

Comment:
Why not to use smth like `lastBlockGasUsed > gasLimit / 10` here?

Reply:
Yes It is same, @tariqsalam requested to convert it to using multiplication as `gasLimit/10` will be integer division

Reply:
actually gasLimit/10 will not be a problem because gasLimit is a constant and a multiple of 10

Reply:
anyways it is same.. will that be a problem?

Reply:
Yes,  it makes it less readable and increases error posibility. Remove this function please